libraries/core/src/main/java/com/paritytrading/philadelphia/FIXMessage.java

```diff
@@ -146,7 +159,19 @@ public FIXValue valueOf(int tag) {
      *   specified tag
      */
     public int indexOf(int tag) {
-        for (int i = 0; i < count; i++) {
+        return indexOf(tag, 0);
+    }
```

Comment:
I would retain the existing `indexOf` body to to avoid indirection in the most common case.